code/game/machinery/suit_storage_unit.dm

```diff
@@ -979,8 +979,8 @@
 	return
 
 //There HAS to be a less bloated way to do this. TODO: some kind of table/icon name coding? ~Z
-/obj/machinery/suit_cycler/proc/apply_paintjob()
-
+//hold onto your hat, this sumbitch just got streamlined -KK
+/obj/machinery/suit_cycler/proc/apply_paintjob(var/obj/item/clothing/head/helmet/parent_helmet as obj,var/obj/item/clothing/suit/space/parent_suit as obj)	

```

Comment:
Question:  Looks like the two parameter values passed in ignored and overwritten in the switch statement below yes?

Reply:
Incorporated into the latest update with a minor adjustment.